pkgs/applications/graphics/processing/default.nix

```diff
@@ -1,73 +1,78 @@
+let
+  vaqua = fetchurl {
+    name = "VAqua9.jar";
+    url = "https://violetlib.org/release/vaqua/9/VAqua9.jar";
+    sha256 = "cd0b82df8e7434c902ec873364bf3e9a3e6bef8b59cbf42433130d71bf1a779c";
+  };
+
+  jna = fetchurl {
+    name = "jna-5.10.0.zip";
+    url = "https://github.com/java-native-access/jna/archive/5.10.0.zip";
+    sha256 = "B5CakOQ8225xNsk2TMV8CbK3RcsLlb+pHzjaY5JNwg0=";
+  };
+
+  flatlaf = fetchurl {
+    name = "flatlaf-2.4.jar";
+    url = "https://repo1.maven.org/maven2/com/formdev/flatlaf/2.4/flatlaf-2.4.jar";
+    sha256 = "NVMYiCd+koNCJ6X3EiRx1Aj+T5uAMSJ9juMmB5Os+zc=";
+  };
 
+in
 stdenv.mkDerivation rec {
   pname = "processing";
-  version = "3.5.4";
+  version = "1286-4.0.1";
 
   src = fetchFromGitHub {
     owner = "processing";
-    repo = "processing";
-    rev = "processing-0270-${version}";
-    sha256 = "0cvv8jda9y8qnfcsziasyv3w7h3w22q78ihr23cm4an63ghxci58";
+    repo = "processing4";
+    rev = "processing-${version}";
+    sha256 = "sha256-U6qYHpkf7yCWnrl0GfDGgIrCN62AyNarRy/A1iUDlUQ=";
   };
 
-  patches = [
-    (fetchpatch {
-      name = "oraclejdk-8u281-compat.patch";
-      url = "https://github.com/processing/processing/commit/7e176876173c93e3a00a922e7ae37951366d1761.patch";
-      sha256 = "g+zwpoIVgw7Sp6QWW3vyPZ/fKHk+o/YCY6xnrX8IGKo=";
-    })
-  ];
+  nativeBuildInputs = [ ant unzip makeWrapper wrapGAppsHook ];
+  buildInputs = [ jdk javaPackages.jogl_2_3_2 ant rsync ffmpeg batik ];
 
-  nativeBuildInputs = [ ant rsync makeWrapper ];
-  buildInputs = [ jdk ];
+  dontWrapGApps = true;
 
   buildPhase = ''
-    # use compiled jogl to avoid patchelf'ing .so files inside jars
-    rm core/library/*.jar
-    cp ${javaPackages.jogl_2_3_2}/share/java/*.jar core/library/
-
-    # do not download a file during build
-    ${xmlstarlet}/bin/xmlstarlet ed --inplace -P -d '//get[@src="http://download.processing.org/reference.zip"]' build/build.xml
-    install -D -m0444 ${fetchurl {
-                          # Use archive.org link for reproducibility until the following issue is fixed:
-                          # https://github.com/processing/processing/issues/5711
-                          url = "https://web.archive.org/web/20200406132357/https://download.processing.org/reference.zip";
-                          sha256 = "093hc7kc9wfxqgf5dzfmfp68pbsy8x647cj0a25vgjm1swi61zbi";
-                        }
-                       } ./java/reference.zip
-
-    # suppress "Not fond of this Java VM" message box
-    substituteInPlace app/src/processing/app/platform/LinuxPlatform.java \
-      --replace 'Messages.showWarning' 'if (false) Messages.showWarning'
-
-    ( cd build
-      substituteInPlace build.xml --replace "jre-download," ""  # do not download jre1.8.0_144
-      mkdir -p linux/jre1.8.0_144                               # fake dir to avoid error
-      ant build )
+    echo "tarring jdk"
+    tar --checkpoint=10000 -czf build/linux/jdk-17.0.4-amd64.tgz ${jdk}
```

Comment:
Processing's build script is looking for `build/linux/jdk-17.0.4-amd64.tgz` but I don't think it matters much which version of Java that file actually contains.

It may be more resilient if we fetched the exact version from https://github.com/adoptium/temurin17-binaries/releases/download/jdk-17.0.4%2B8/OpenJDK17U-jdk_x64_linux_hotspot_17.0.4_8.tar.gz.

Reply:
I don't think we want to fetch the exact version.

Reply:
Sounds good, I'll keep it as-is.